source/MaterialXCore/Interface.cpp

```diff
@@ -43,6 +43,15 @@ const std::unordered_map<string, size_t> PortElement::CHANNELS_PATTERN_LENGTH =
     { "vector4", 4 }
 };
 
+// Structure needed for ValueElement set comparison
+struct ValueElementCompare
```

Comment:
Seems strange to have this compare declared here. Can we localize it by setting a lambda when declaring the test. Seems to be the C++11 way vs this way which is the older way to do this.